lib/internal/test_runner/harness.js

```diff
@@ -52,14 +53,16 @@ function createProcessEventHandler(eventName, rootTest) {
       throw err;
     }
 
-    // Check if this error is coming from a test. If it is, fail the test.
-    if (!test || test.finished) {
+    // Check if this error is coming from a test or test hook. If it is, fail the test.
+    if (!test || test.finished || test instanceof TestHook) {
       // If the test is already finished or the resource that created the error
       // is not mapped to a Test, report this as a top level diagnostic.
       let msg;
 
       if (test) {
-        msg = `Warning: Test "${test.name}" generated asynchronous ` +
+        const name = test instanceof TestHook ? `Test Hook "${test.hookType}"` : `Test "${test.name}"`;
+        const relPath = relative(process.cwd(), test.loc.file);
+        msg = `Warning: ${name} at ${relPath}:${test.loc.line}:${test.loc.column} generated asynchronous ` +
           'activity after the test ended. This activity created the error ' +
           `"${err}" and would have caused the test to fail, but instead ` +
           `triggered an ${eventName} event.`;
```

Comment:
There is another warning on line 70 that we should change to an error. We should also remove the part that says "after the test ended" since we have seen that it can happen before a test even starts.

Reply:
> We should also remove the part that says "after the test ended" since we have seen that it can happen before a test even starts.

I will keep this comment out of the context of this PR. And perhaps we can open another PR and add such test case